classes/sota_sanity.bbclass

```diff
@@ -10,6 +24,40 @@ def sota_check_required_variables(status, d):
         if not d.getVar(var):
             status.addresult("%s should be set in your local.conf.\n" % var)
 
+def sota_check_variables_validity(status, d):
+    if re.match("^[a-zA-Z0-9_-]*$", d.getVar("OSTREE_BRANCHNAME")) is None:
+        status.addresult("OSTREE_BRANCHNAME Should only contain characters from the character set [a-zA-Z0-9_-].\n")
+    if re.match("^[a-zA-Z0-9_-]*$", d.getVar("SOTA_HARDWARE_ID")) is None:
+        status.addresult("SOTA_HARDWARE_ID Should only contain characters from the character set [a-zA-Z0-9_-].\n")
+    if d.getVar("SOTA_CLIENT_FEATURES") not in ("hsm", "secondary-network", ""):
+        status.addresult("SOTA_CLIENT_FEATURES should be set to hsm or secondary-network.\n")
+    if d.getVar("SOTA_CLIENT_PROV") not in ("aktualizr-shared-prov", "aktualizr-device-prov", "aktualizr-device-prov-hsm", ""):
+        status.addresult("Valid options for SOTA_CLIENT_PROV are aktualizr-shared-prov, aktualizr-device-prov, and aktualizr-device-prov-hsm.\n")
+    if re.match("^(https?|ftp|file)://.+$", d.getVar("GARAGE_TARGET_URL")) is None:
+        status.addresult("GARAGE_TARGET_URL is set to a bad url.\n")
+    if re.match("^^((([0-9]{3}[1-9]|[0-9]{2}[1-9][0-9]{1}|[0-9]{1}[1-9][0-9]{2}|[1-9][0-9]{3})-(((0[13578]|1[02])-(0[1-9]|[12][0-9]|3[01]))|((0[469]|11)-(0[1-9]|[12][0-9]|30))|(02-(0[1-9]|[1][0-9]|2[0-8]))))|((([0-9]{2})(0[48]|[2468][048]|[13579][26])|((0[48]|[2468][048]|[3579][26])00))-02-29))T([0-1]?[0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])Z$", d.getVar("GARAGE_TARGET_EXPIRES")) is None:
+        status.addresult("GARAGE_TARGET_EXPIRES expects a valid UTC instant.\n")
+    if re.match("^([0-9]{1,4})Y(?:1[0-2]|[1-9])M(?:[1-9]|([1-2][0-9])|3[0-1])D$", d.getVar("GARAGE_TARGET_EXPIRE_AFTER")) is None:
+        status.addresult("Specify the number of years, months and days like \"1Y3M5D\" for GARAGE_TARGET_EXPIRE_AFTER.\n")
+    if re.match("^[1-9][0-9]*$", d.getVar("SOTA_POLLING_SEC")) is None:
+        status.addresult("SOTA_POLLING_SEC should be an integer.\n")
+    if re.match(r"^[a-zA-Z0-9_-\:\/\\]*$", d.getVar("OSTREE_REPO")) is None:
+        status.addresult("OSTREE_REPO is not set correctly. Path to your OSTree repository is invalid.\n")
+    path = d.getVar("SOTA_SECONDARY_CONFIG")
+    if not os.path.exists(path):
+        status.addresult("SOTA_SECONDARY_CONFIG is not set correctly.The file containing JSON configuration for secondaries does not exist.\n")
+    path = d.getVar("SOTA_PACKED_CREDENTIALS")
+    if not os.path.exists(path):
+        status.addresult("SOTA_PACKED_CREDENTIALS is not set correctly.The zipped credentials file does not exist.\n")
```

Comment:
For both `SOTA_SECONDARY_CONFIG` and `SOTA_PACKED_CREDENTIALS`, leaving them unset is also a valid option. And less critical, but there should be spaces after the periods between the two sentences in the error message.

Reply:
> leaving them unset
Is setting `SOTA_PACKED_CREDENTIALS` to  "" a valid option?

Reply:
That's worth double-checking. I know that garage-push and garage-sign won't work without credentials, but I think you should be able to bitbake an image without credentials. If you're using device credential provisioning and you don't want to push the image to our backend for some reason, you shouldn't need credentials. So I think empty is valid, even if not recommended.